buf/validate/internal/extra_func.cc

```diff
@@ -220,6 +220,130 @@ cel::CelValue isIP(google::protobuf::Arena* arena, cel::CelValue::StringHolder l
   return isIPvX(arena, lhs, cel::CelValue::CreateInt64(0));
 }
 
+/**
+ * IP Prefix Validation
+ */
+bool IsIpv4Prefix(const std::string_view to_validate, bool strict) {
+  std::vector<std::string> split = absl::StrSplit(to_validate.data(), '/');
+  if (split.size() != 2) {
+    return false;
+  }
+  std::string ip = split[0];
+  std::string prefixlen = split[1];
+
+  // validate ip
+  if (!IsIpv4(ip)) {
+    return false;
+  }
+
+  // validate prefixlen
+  if (prefixlen.empty()) {
+    return false;
+  }
+  int prefixlen_int;
+  if (!absl::SimpleAtoi(prefixlen, &prefixlen_int)) {
+    return false;
+  }
+  if (prefixlen_int < 0 || prefixlen_int > 32) {
+    return false;
+  }
+
+  // check host part is all-zero if strict
+  if (strict) {
+    struct sockaddr_in sa;
+    if (inet_pton(AF_INET, ip.c_str(), &sa.sin_addr) != 1) {
+      return false;
+    }
+    struct in_addr mask;
+    mask.s_addr = htonl(std::pow(2, 32 - prefixlen_int) - 1);
```

Comment:
I believe that `std::pow(2, 32 - prefixlen_int)` can be replaced with `1 << (32-prefixlen_int)` (or perhaps `1ull << (32-prefixlen_int)` if needed to avoid the integer overflow.)